### PR TITLE
Tck reader

### PR DESCRIPTION
**Pull request by FrancescoGobbi**
Added features to read .tck files.
In the fiber reading the affine matrix is used to allow the correct overlap between the COMMIT outputs between .trk and .tck

### CODE REVIEW

File: commit/trk2dictionary/trk2dictionary.pyx [hunk 1]
```diff
@@ -8,19 +8,22 @@ import nibabel
...
+        int n_properties, float fiber_shiftX, float fiber_shiftY, float fiber_shiftZ, int points_to_skip, float min_seg_len,
         float* ptrPEAKS, int Np, float vf_THR, int ECix, int ECiy, int ECiz,
         float* _ptrMASK, float* ptrTDI, char* path_out, int c, double* ptrAFFINE,
-        int nBlurRadii, double blurSigma, double* ptrBlurRadii, int* ptrBlurSamples, double* ptrBlurWeights
+        int nBlurRadii, double blurSigma, double* ptrBlurRadii, int* ptrBlurSamples, double* ptrBlurWeights,  float* ptrArrayInvM
     ) nogil
 
 
-cpdef run( filename_trk, path_out, filename_peaks = None, filename_mask = None, do_intersect = True,
+cpdef run( filename_tractogram, path_out, TCK_ref_image = None, filename_peaks = None, filename_mask = None, do_intersect = True,
```
Comment (MarioOcampo) [inline commit/trk2dictionary/trk2dictionary.pyx]: filename_tractogram is not compatible with previous scripts. 
Documentation and tutorials have different variable name

File: commit/trk2dictionary/trk2dictionary.pyx [hunk 2]
```diff
@@ -8,19 +8,22 @@ import nibabel
 from os.path import join, exists
 from os import makedirs, remove
 import time
+import os.path
```
Comment (MarioOcampo) [inline commit/trk2dictionary/trk2dictionary.pyx]: Use the `from os.path` of line 8 to import the function `splitext`

File: commit/trk2dictionary/trk2dictionary.pyx [hunk 3]
```diff
@@ -157,21 +162,85 @@ cpdef run( filename_trk, path_out, filename_peaks = None, filename_mask = None,
 
     # fiber-tracts from .trk
     print( '\t\t* tractogram' )
-    try :
-        _, trk_hdr = nibabel.trackvis.read( filename_trk )
+    
+    extension = os.path.splitext(filename_tractogram)[1]  #take extension of file
```
Comment (MarioOcampo) [inline commit/trk2dictionary/trk2dictionary.pyx]: Remove `os.path`

File: commit/trk2dictionary/trk2dictionary_c.cpp [hunk 4]
```diff
@@ -146,10 +166,33 @@ int trk2dictionary(
...
+    
+    float affine[4][4];
+    if (!isTRK)  {//.tck
+        //ricreate affine matrix
+        int k = 0;
+        for(int i=0; i<4; i++) {
+            for (int j=0; j<4; j++) {
+                affine[i][j] = VetAffine[k];
+                k++;
+                //printf("%f ", affine[i][j]);
```
Comment (MarioOcampo) [inline commit/trk2dictionary/trk2dictionary_c.cpp]: Remove commented functions (printf)

File: commit/trk2dictionary/trk2dictionary.pyx (hunk 1 above)
Comment (MarioOcampo) [reply commit/trk2dictionary/trk2dictionary.pyx]: Change `filename_tractogram = None, path_out = None` and add as last parameter `filename_trk = None`.

Then, check that [`path_out` is not None]  AND  [at least one of `filename_tractogram` or `filename_tractogram` is not None]. If the previous is not satisfy, rise and error. 

If `filename_trk`is used, show (print) a warning, this parameter will be deprecated. 

If `filename_tractogram` and `filename_trk` are used, show a message like:  filename_tractogram will be used, filename_trk will not be considered"

File: commit/trk2dictionary/trk2dictionary.pyx [hunk 5]
```diff
@@ -35,6 +35,8 @@ cpdef run( filename_tractogram, path_out, TCK_ref_image = None, filename_peaks =
     ----------
     filename_tractogram : string
         Path to the .trk or .tck file containing the tractogram to convert.
+        
+    filename_trk : DEPRECATED
```
Comment (daducci) [inline commit/trk2dictionary/trk2dictionary.pyx]: Should be: "filename_trk : DEPRECATED. Use filename_tractogram instead."

(or something similar)

File: commit/trk2dictionary/trk2dictionary.pyx [hunk 6]
```diff
@@ -22,7 +22,7 @@ cdef extern from "trk2dictionary_c.cpp":
     ) nogil
 
 
-cpdef run( filename_tractogram, path_out, TCK_ref_image = None, filename_peaks = None, filename_mask = None, do_intersect = True,
+cpdef run( filename_tractogram = None, filename_trk = None, path_out = None, TCK_ref_image = None, filename_peaks = None, filename_mask = None, do_intersect = True,
```
Comment (daducci) [inline commit/trk2dictionary/trk2dictionary.pyx]: @MarioOcampo , shouldn't this be `filename_tractogram = None, path_out = None, filename_trk = None` instead? In the way it is implemented, is the positional passing of parameters backward compatible?

**Comment by FrancescoGobbi** [comment]
Ok, I just committed to edit.
Is the rest okay?
Do we keep "filename_trk" and "filename_tractogram" to None?

Francesco

Il giorno mar 26 nov 2019 alle ore 14:24 Alessandro Daducci <
notifications@github.com> ha scritto:

File: commit/trk2dictionary/trk2dictionary.pyx (hunk 6 above)
Comment (MarioOcampo) [reply commit/trk2dictionary/trk2dictionary.pyx]: Yes. it should be at the beginning  `filename_tractogram = None, path_out = None` and as last parameter `filename_trk = None`.

File: commit/trk2dictionary/trk2dictionary.pyx [hunk 7]
```diff
@@ -5,34 +5,39 @@ import cython
...
+        int nBlurRadii, double blurSigma, double* ptrBlurRadii, int* ptrBlurSamples, double* ptrBlurWeights,  float* ptrArrayInvM
     ) nogil
 
 
-cpdef run( filename_trk, path_out, filename_peaks = None, filename_mask = None, do_intersect = True,
+cpdef run( filename_tractogram = None, path_out = None, filename_peaks = None, filename_mask = None, do_intersect = True,
     fiber_shift = 0, points_to_skip = 0, vf_THR = 0.1, peaks_use_affine = False,
     flip_peaks = [False,False,False], min_seg_len = 1e-3, gen_trk = True,
-    blur_radii = [], blur_samples = [], blur_sigma = 1.0
+    blur_radii = [], blur_samples = [], blur_sigma = 1.0, filename_trk = None, TCK_ref_image = None
```
Comment (daducci) [inline commit/trk2dictionary/trk2dictionary.pyx]: There's no documentation for the parameter `TCK_ref_image`

File: commit/trk2dictionary/trk2dictionary.pyx [hunk 8]
```diff
@@ -157,21 +167,99 @@ cpdef run( filename_trk, path_out, filename_peaks = None, filename_mask = None,
 
     # fiber-tracts from .trk
     print( '\t\t* tractogram' )
-    try :
-        _, trk_hdr = nibabel.trackvis.read( filename_trk )
+    
+    if (path_out is None):
+        raise IOError( 'Path out not defined' )
```
Comment (daducci) [inline commit/trk2dictionary/trk2dictionary.pyx]: Shall we use `IOError` or `RuntimeError` as before? I don't know what the best strategy is

Comment (MarioOcampo) [reply commit/trk2dictionary/trk2dictionary.pyx]: I think `RuntimeError`.

**Pull request merged**